### PR TITLE
Alpha dev 01 bump version

### PR DESCRIPTION
**Pull request by atoomic**
This is still a work in a progress to reach objective 01 #152

### CODE REVIEW

**Comment by jkeenan** [comment]
@atoomic, as of Sat Jul 25 20:56:13 UTC 2020, at commit dc674470e1, branch alpha-dev-01-bump-version, on an unthreaded build on Linux, we still have failures in Storable, in `t/comp/use.t` and in `t/run/switches.t` (the latter contrary to what you indicated in channel).

```
comp/use.t (Wstat: 0 Tests: 84 Failed: 11)
  Failed tests:  10, 12-16, 19-20, 58-60
Files=1, Tests=84,  0 wallclock secs ( 0.02 usr +  0.00 sys =  0.02 CPU)
Result: FAIL
```
```
$ cd t;TEST_JOBS=1 ./perl harness run/switches.t ; cd -
run/switches.t .. 1/135 # Failed test 27 - -V looks okay at run/switches.t line 270
#      got 'Summary of my perl5 (revision 7 version 0 subversion 0) configuration:\n  Commit id: 070556c52366e9881d95787cdbef40fcf8b6663d\n  Platform:\n    osname=linux\n    osvers=5.1.17-x86_64-linode128\n    archname=x86_64-linux\n    uname=\'linux li796-253 5.1.17-x86_64-linode128 #1 smp preempt wed jul 10 17:11:49 utc 2019 x86_64 gnulinux \'\n    config_args=\'-des -Dusedevel\'\n    hint=recommended\n    useposix=true\n    d_sigaction=define\n    useithreads=undef\n    usemultiplicity=undef\n    use64bitint=define\n    use64bitall=define\n    uselongdouble=undef\n    usemymalloc=n\n    default_inc_excludes_dot=define\n    bincompat5005=undef\n  Compiler:\n    cc=\'cc\'\n    ccflags =\'-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2\'\n    optimize=\'-O2\'\n    cppflags=\'-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include\'\n    ccversion=\'\'\n    gccversion=\'4.9.2\'\n    gccosandvers=\'\'\n    intsize=4\n    longsize=8\n    ptrsize=8\n    doublesize=8\n    byteorder=12345678\n    doublekind=3\n    d_longlong=define\n    longlongsize=8\n    d_longdbl=define\n    longdblsize=16\n    longdblkind=3\n    ivtype=\'long\'\n    ivsize=8\n    nvtype=\'double\'\n    nvsize=8\n    Off_t=\'off_t\'\n    lseeksize=8\n    alignbytes=8\n    prototype=define\n  Linker and Libraries:\n    ld=\'cc\'\n    ldflags =\' -fstack-protector-strong -L/usr/local/lib\'\n    libpth=/usr/local/lib /usr/lib/gcc/x86_64-linux-gnu/4.9/include-fixed /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib\n    libs=-lpthread -lnsl -ldb -ldl -lm -lcrypt -lutil -lc\n    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc\n    libc=libc-2.19.so\n    so=so\n    useshrplib=false\n    libperl=libperl.a\n    gnulibc_version=\'2.19\'\n  Dynamic Linking:\n    dlsrc=dl_dlopen.xs\n    dlext=so\n    d_dlsymun=undef\n    ccdlflags=\'-Wl,-E\'\n    cccdlflags=\'-fPIC\'\n    lddlflags=\'-shared -O2 -L/usr/local/lib -fstack-protector-strong\'\n\n\nCharacteristics of this binary (from libperl): \n  Compile-time options:\n    HAS_TIMES\n    PERLIO_LAYERS\n    PERL_COPY_ON_WRITE\n    PERL_DONT_CREATE_GVSV\n    PERL_MALLOC_WRAP\n    PERL_OP_PARENT\n    PERL_PRESERVE_IVUV\n    PERL_USE_DEVEL\n    USE_64_BIT_ALL\n    USE_64_BIT_INT\n    USE_LARGE_FILES\n    USE_LOCALE\n    USE_LOCALE_COLLATE\n    USE_LOCALE_CTYPE\n    USE_LOCALE_NUMERIC\n    USE_LOCALE_TIME\n    USE_PERLIO\n    USE_PERL_ATOF\n  Built under linux\n  Compiled at Jul 25 2020 14:32:05\n  %ENV:\n    PERLBREW_BASHRC_VERSION=\"0.80\"\n    PERLBREW_HOME=\"/home/jkeenan/.perlbrew\"\n    PERLBREW_MANPATH=\"/home/jkeenan/perl5/perlbrew/perls/perl-5.28.0/man\"\n    PERLBREW_PATH=\"/home/jkeenan/perl5/perlbrew/bin:/home/jkeenan/perl5/perlbrew/perls/perl-5.28.0/bin\"\n    PERLBREW_PERL=\"perl-5.28.0\"\n    PERLBREW_ROOT=\"/home/jkeenan/perl5/perlbrew\"\n    PERLBREW_VERSION=\"0.80\"\n    PERLEXE=\"/home/jkeenan/gitwork/perl-atoomic/perl\"\n    PERL_CORE=\"1\"\n    PERL_WORKDIR=\"/home/jkeenan/gitwork/perl\"\n  @INC:\n    ../lib\n    .\n    /usr/local/lib/perl5/site_perl/7.0.0/x86_64-linux\n    /usr/local/lib/perl5/site_perl/7.0.0\n    /usr/local/lib/perl5/7.0.0/x86_64-linux\n    /usr/local/lib/perl5/7.0.0\n'
# expected /(?^:\ASummary of my perl7 .*configuration:)/
Feature bundle "5.0" is not supported by Perl 7.0.0 at -e line 0.
BEGIN failed--compilation aborted.
# Failed test 130 - -E say at run/switches.t line 690
#      got ""
# expected "Hello, world!\n"
Feature bundle "5.0" is not supported by Perl 7.0.0 at -e line 0.
BEGIN failed--compilation aborted.
# Failed test 131 - -E ~~ at run/switches.t line 696
#      got ""
# expected "Hello, world!\n"
Feature bundle "5.0" is not supported by Perl 7.0.0 at -e line 0.
BEGIN failed--compilation aborted.
# Failed test 132 - -E given at run/switches.t line 701
#      got ""
# expected "Hello, world!\n"
Feature bundle "5.0" is not supported by Perl 7.0.0 at -e line 0.
BEGIN failed--compilation aborted.
# Failed test 133 - -E works outside of the block created by -n at run/switches.t line 707
#      got ""
# expected "affe\n"
Feature bundle "5.0" is not supported by Perl 7.0.0 at -e line 0.
BEGIN failed--compilation aborted.
run/switches.t .. 134/135 # Failed test 134 - -E does not enable strictures at run/switches.t line 712
#      got ""
# expected "Hello, world!\n"
run/switches.t .. Failed 6/135 subtests 
	(less 4 skipped subtests: 125 okay)

Test Summary Report
-------------------
run/switches.t (Wstat: 0 Tests: 135 Failed: 6)
  Failed tests:  27, 130-134
Files=1, Tests=135,  1 wallclock secs ( 0.03 usr  0.01 sys +  0.38 cusr  0.10 csys =  0.52 CPU)
Result: FAIL
```

**Comment by jkeenan** [comment]
There were several porting things that needed regeneration.  After I did that, `t/run/switches.t` once again runs clean.
```
$ cd t;TEST_JOBS=1 ./perl harness run/switches.t; cd -
run/switches.t .. ok       
All tests successful.
Files=1, Tests=135,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.19 cusr  0.27 csys =  0.48 CPU)
Result: PASS
```
That leaves the Storable tests, `t/comp/use.t` ... and threaded builds :-(

**Comment by jkeenan** [comment]
In `dist/Storable/t/downgrade.t`, **some** of the individual test failures occur because we are not capturing an exception like:
```
Modification of non-creatable hash value attempted, subscript "rules" at 
  t/downgrade.t line ___
```
In turn, **some** of these failures are associated with a warning which is now being thrown:
```
Use of uninitialized value $value in reverse at t/downgrade.t line ___, <DATA> line 30.
```
... where ___ is 119, 133 or 136.  If you examine those lines, you can see that in Perl 5, the possible `undef` value of `$value` at those points was never taken into consideration.

(I'm not exactly sure why I'm now seeing that warning.  We haven't yet implemented warnings-on-default, have we?)

**Comment by atoomic** [comment]
We have not implemented `strict` and `warnings` as default at this point
the errors we are seeing seems to come from the modification of a read only variable
which probably indicates a deeper problem than Storable
I spent some time but need to continue investigating

File: INSTALL [hunk 1]
```diff
@@ -615,7 +615,7 @@ The directories set up by Configure fall into three broad categories.
 
 =item Directories for the perl distribution
 
-By default, Configure will use the following directories for 5.32.0.
+By default, Configure will use the following directories for 5.33.0.
```
Comment (tonycoz) [inline INSTALL]: perl7? (a few more cases in this file)

File: NetWare/Makefile [hunk 2]
```diff
@@ -86,7 +86,7 @@ NLM_VERSION    = 3,20,0
 
 
 # Here comes the CW tools - TO BE FILLED TO BUILD WITH CW -
-MODULE_DESC     = "Perl 5.32.0 for NetWare"
+MODULE_DESC     = "Perl 5.33.0 for NetWare"
```
Comment (tonycoz) [inline NetWare/Makefile]: 7 here and for INST_VER.

File: MANIFEST [hunk 3]
```diff
@@ -834,7 +834,7 @@ cpan/ExtUtils-MakeMaker/t/META_for_testing.yml				test data for MakeMaker
 cpan/ExtUtils-MakeMaker/t/META_for_testing_tricky_version.yml				test data for MakeMaker
 cpan/ExtUtils-MakeMaker/t/metafile_data.t			See if META.yml handling works
 cpan/ExtUtils-MakeMaker/t/metafile_file.t			See if META.yml handling works
-cpan/ExtUtils-MakeMaker/t/min_perl_version.t			Test the MIN_PERL_VERSION argument to WriteMakefile.
+cpan/ExtUtils-MakeMaker/t/min_perl_version.t			Test the MIN_PERL_CORE_MINOR argument to WriteMakefile.
```
Comment (tonycoz) [inline MANIFEST]: pretty sure this is wrong.  MIN_PERL_VERSION accepts a fill version number and doesn't require a change, and no changes appear to have been made to `cpan/ExtUtils-MakeMaker/*`

File: Makefile.SH [hunk 4]
```diff
@@ -589,7 +589,7 @@ esac
 
 $spitshell >>$Makefile <<'!NO!SUBS!'
 
-perltoc_pod_prereqs = extra.pods pod/perl5320delta.pod pod/perlapi.pod pod/perlintern.pod pod/perlmodlib.pod pod/perluniprops.pod
+perltoc_pod_prereqs = extra.pods pod/perl5330delta.pod pod/perlapi.pod pod/perlintern.pod pod/perlmodlib.pod pod/perluniprops.pod
```
Comment (tonycoz) [inline Makefile.SH]: Is this perl 7.0.0 or perl 5.33.0 ?  This changeset appears to be confused about that.

**Comment by toddr** [comment]
I don't consider this done without first doing `perl -pi -e 's/5.33.0/7.0.0/'` Just looking at the goal, it says bump the version. the version has not been bumped in all the places and we may regret this incompleteness later.

File: MANIFEST (hunk 3 above)
Comment (atoomic) [reply MANIFEST]: you are right this need to be restored to the previous definition

File: Makefile.SH (hunk 4 above)
Comment (atoomic) [reply Makefile.SH]: yeap you are right, need to replace all `s{5.33.0}{7.0.0}g`

File: INSTALL (hunk 1 above)
Comment (atoomic) [reply INSTALL]: indeed going to make these changes, was concerned about the noise of such a change in a first iteration 
but will do this in a later commit

File: MANIFEST (hunk 3 above)
Comment (atoomic) [reply MANIFEST]: fixed by 65ccb1427af285b0d15560552505b8ef5383802e

File: Makefile.SH (hunk 4 above)
Comment (atoomic) [reply Makefile.SH]: `pod/perl700delta.pod` bump made via 60b0a0d1a3e783496ff1ca4e159c0c2e8013a05d

**Comment by atoomic** [comment]
@jkeenan this is ready for final review/test/merge

**Comment by jkeenan** [comment]
I locally merged the `alpha-dev-01-bump-version` branch into the `alpha` branch today, Aug 07 2020.
```
commit 800f84ad070df976fd5e07a60968328a254c15f6 (HEAD -> alpha, tag: alpha-01, upstream/alpha-dev-02-strict, upstream/alpha, origin/alpha-dev-02-strict, origin/alpha, alpha-dev-02-strict)
Author:     ☢ ℕicolas ℝ <nicolas@atoomic.org>
AuthorDate: Mon Aug 3 17:05:56 2020
Commit:     James E Keenan <jkeenan@cpan.org>
CommitDate: Fri Aug 7 07:55:36 2020
```
This completes the work on [Objective 1](https://github.com/atoomic/perl/issues/152).

Thank you very much.
Jim Keenan

**Pull request closed without merging**